lib/rules/no-misspelled-properties.js

```diff
@@ -18,38 +15,133 @@ var getCombinedList = function (props, extras) {
+};
+
 module.exports = {
   'name': 'no-misspelled-properties',
   'defaults': {
     'extra-properties': []
   },
   'detect': function (ast, parser) {
     var result = [];
+    var toSkip = 0;
+    var propertyList = getCombinedList(properties, parser.options['extra-properties']);
 
-    ast.traverseByType('property', function (node) {
-      if (node.first().is('ident')) {
-        var curProperty = node.first().content,
-            propertyList = getCombinedList(properties, parser.options['extra-properties']);
+    ast.traverseByType('declaration', function (node) {
+      var prop = node.first('property');
+      var containsInterp = prop.contains('interpolation');
+      // If we've already checked declarations in a multiline we can skip those decs here
+      if (toSkip) {
+        toSkip--;
+        return !toSkip;
+      }
+      // make sure our first node within our property is an ident
+      if (!prop.first() || !prop.first().is('ident')) {
+        return false;
+      }
 
-        if (curProperty.charAt(0) === '-') {
-          curProperty = helpers.stripPrefix(curProperty);
-        }
+      var curProperty = prop.first() && prop.first().is('ident') && prop.first('ident').content;
+      var value = node.first('value');
+      var fullProperties = [];
 
-        if (helpers.isPartialStringMatch(curProperty, propertyList)) {
-          return false;
-        }
-
-        if (curProperty.length > 0) {
-          result = helpers.addUnique(result, {
-            'ruleId': parser.rule.name,
-            'line': node.start.line,
-            'column': node.start.column,
-            'message': 'Property `' + curProperty + '` appears to be spelled incorrectly',
-            'severity': parser.severity
-          });
+      if (value.contains('block')) {
+        // encountered a multiline property, we should build the property list here
+        fullProperties = buildPartialProperty(
+          value.first('block'),
+          {
+            name: curProperty,
+            line: prop.first('ident').start.line,
+            col: prop.first('ident').start.column
+          },
+          0
+        );
+      }
+      // If we have multiline properties
+      if (fullProperties.length) {
+        fullProperties.forEach(function (constrProp) {
+          // Add the number of propertiy declarations we've already checked here so we can skip them
+          toSkip += constrProp.propsEncountered;
+          // Check if the property exists in our list
+          if (propertyList.indexOf(constrProp.name) === -1) {
+            result = helpers.addUnique(result, {
+              'ruleId': parser.rule.name,
+              'line': constrProp.line,
+              'column': constrProp.col,
+              'message': 'Property `' + constrProp.name + '` appears to be spelled incorrectly',
+              'severity': parser.severity
+            });
+          }
+        });
+      }
+      else {
+        if (curProperty && curProperty.length > 0) {
```

Comment:
Could combine the else if and just run with `...curProperty.length)`

Reply:
curProperty 'could' technically be undefined here. Would rather leave the check just in case as we don't wanna get length of undefined. Can remove the `> 0` though as that's unnecessary

Reply:
Aye, sorry the `> 0` is what I was talking about :+1: